packages/zipkin-instrumentation-request/src/wrapRequest.js

```diff
@@ -1,33 +1,39 @@
 const {Request, Annotation} = require('zipkin');
 
 function wrapRequest(request, {tracer, serviceName = 'unknown', remoteServiceName}) {
-  return request.defaults((options, callback) => {
-    tracer.scoped(() => {
-      tracer.setId(tracer.createChildId());
-      const traceId = tracer.id;
+  return request.defaults((options, callback) => tracer.scoped(() => {
+    tracer.setId(tracer.createChildId());
+    const traceId = tracer.id;
 
-      const wrappedOptions = Request.addZipkinHeaders(options, tracer.id);
-      tracer.recordServiceName(serviceName);
-      const method = wrappedOptions.method || 'GET';
-      tracer.recordRpc(method.toUpperCase());
-      tracer.recordBinary('http.url', wrappedOptions.uri || wrappedOptions.url);
-      tracer.recordAnnotation(new Annotation.ClientSend());
-      if (remoteServiceName) {
-        tracer.recordAnnotation(new Annotation.ServerAddr({
-          serviceName: remoteServiceName
-        }));
-      }
+    const wrappedOptions = Request.addZipkinHeaders(options, tracer.id);
+    const method = wrappedOptions.method || 'GET';
 
-      const wrappedCallback = (error, response, body) => {
-        tracer.setId(traceId);
-        tracer.recordBinary('http.status_code', response.statusCode.toString());
-        tracer.recordAnnotation(new Annotation.ClientRecv());
-        return callback(error, response, body);
-      };
+    tracer.recordServiceName(serviceName);
+    tracer.recordRpc(method.toUpperCase());
+    tracer.recordBinary('http.url', wrappedOptions.uri || wrappedOptions.url);
+    tracer.recordAnnotation(new Annotation.ClientSend());
+    if (remoteServiceName) {
+      tracer.recordAnnotation(new Annotation.ServerAddr({
+        serviceName: remoteServiceName
+      }));
+    }
 
-      request(wrappedOptions, wrappedCallback);
-    });
-  });
+    const recordResponse = (response) => {
+      tracer.setId(traceId);
+      tracer.recordBinary('http.status_code', response.statusCode.toString());
+      tracer.recordAnnotation(new Annotation.ClientRecv());
+    };
+
+    const recordError = (error) => {
+      tracer.setId(traceId);
+      tracer.recordBinary('request.error', error.toString());
```

Comment:
ps sometimes we run into things that aren't right in other integrations.. better to correct new ones than follow bad practice. even better to raise another PR to correct the others :)